src/java.base/share/classes/java/io/Console.java

```diff
@@ -562,37 +594,23 @@ public Console console() {
                 }
                 return null;
             }
-
-            public Charset charset() {
-                // This method is called in sun.security.util.Password,
-                // cons already exists when this method is called
-                return cons.cs;
-            }
         });
```

Comment:
Please keep the charset() method and return CHARSET.

I'm looking at a use case that needs to know the platform charset regardless of whether the console exists.
When a process is launched it may be redirected to /dev/tty or a pseudo tty and in that case
a Reader from that stream should be able to use the encoding of the platform.
Its still a work in progress, but it would save some refactoring or duplication later.

Reply:
Would the singleton `Console.cons` be instantiated in your use case? It is created only when isatty() (or Windows' equivalent) in the native code returns true.

Reply:
Not always, for example, if stderr was redirected to a terminal but not stdin and stdout.
The istty check is only true if both stdin and stdout are ttys.

Reply:
Then `charset()` in the shared secret would return `null`. Would that suffice your case?

Reply:
I read lines 575-587 as initializing CHARSET regardless of whether the Console was created.

Reply:
OK, revived the charset() method.